gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.gradle.core;
+
+import com.gradleware.tooling.toolingclient.GradleDistribution;
+
+import java.io.File;
+import java.util.ArrayList;
+
+import org.eclipse.buildship.core.projectimport.ProjectImportConfiguration;
+import org.eclipse.buildship.core.util.gradle.GradleDistributionWrapper;
+import org.eclipse.buildship.core.util.progress.AsyncHandler;
+import org.eclipse.buildship.core.workspace.SynchronizeGradleProjectJob;
+
+/**
+ * @author Andy Wu
+ */
+public class GradleUtil
+{
+
+    public static void importGradleProject( File dir )
+    {
+        ProjectImportConfiguration configuration = new ProjectImportConfiguration();
+        GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
+
+        configuration.setGradleDistribution( from );
+        configuration.setProjectDir( dir );
+        configuration.setApplyWorkingSets( false );
+        configuration.setWorkingSets( new ArrayList<String>() );
```

Comment:
Please add a new ticket about supporting new gradle projects being added to a working set.